File: packages/shared-protocol/src/ts/audit.ts
/**
 * Audit entry shape — mirrors the SQLite row produced by both Rust and Python.
 */
export type AuditAction =
  | 'agent.run.start'
  | 'agent.run.error'
  | 'agent.approval'
  | 'agent.cancel'
  | 'approval.request'
  | 'approval.decision'
  | 'credential.set'
  | 'db.execute'
  | string;

export interface AuditEntry {
  id: number;
  action: AuditAction;
  payload: unknown;
  ts: string;          // RFC 3339
  operator?: string;
  runId?: string;
}

File: packages/shared-protocol/src/protocol/audit.py
"""Audit entry — mirrors the SQLite row from src-tauri + services/agent."""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

AuditAction = Literal[
    "agent.run.start",
    "agent.run.error",
    "agent.approval",
    "agent.cancel",
    "approval.request",
    "approval.decision",
    "credential.set",
    "db.execute",
]


class AuditEntry(BaseModel):
    id: int
    action: AuditAction | str
    payload: object
    ts: datetime
    operator: str | None = None
    run_id: str | None = Field(default=None, alias="runId")
